libyul/Object.h

```diff
@@ -110,7 +110,7 @@ struct Object: public ObjectNode
 	/// pathToSubObject("A1.E2.F3.H4") == {1, 0, 2}
 	/// pathToSubObject("E2.F3.H4") == {1, 0, 2}
 	/// pathToSubObject("A1.E2") == {1}
-	/// The path must not lead to a @a Data object (will throw in that case).
```

Comment:
Why was this even decided so?

Reply:
Good question. In #9235 the way data objects were rejected changed multiple times but the idea that these paths must not refer to data seems to have been there implicitly from the beginning (this is one of the early versions of this docstring: https://github.com/ethereum/solidity/pull/9235#discussion_r442820388) and no one in the PR questioned it.